processor/src/test/java/org/mapstruct/ap/internal/util/StringsTest.java

```diff
@@ -0,0 +1,99 @@
+        assertThat( Strings.capitalize( "capitalize" ) ).isEqualTo( "Capitalize" );
+        assertThat( Strings.capitalize( "AlreadyCapitalized" ) ).isEqualTo( "AlreadyCapitalized" );
+        assertThat( Strings.capitalize( "notCapitalized" ) ).isEqualTo( "NotCapitalized" );
+    }
+
+    @Test
+    public void testDecapitalize() throws Exception {
+        assertThat( Strings.decapitalize( null ) ).isNull();
+        assertThat( Strings.decapitalize( "c" ) ).isEqualTo( "c" );
+        assertThat( Strings.decapitalize( "capitalize" ) ).isEqualTo( "capitalize" );
+        assertThat( Strings.decapitalize( "AlreadyCapitalized" ) ).isEqualTo( "alreadyCapitalized" );
+        assertThat( Strings.decapitalize( "notCapitalized" ) ).isEqualTo( "notCapitalized" );
+    }
+
+    @Test
+    public void testJoin() throws Exception {
+        assertThat( Strings.join( new ArrayList<String>(), "-" ) ).isEqualTo( "" );
+        assertThat( Strings.join( Arrays.asList( "Hello", "World" ), "-" ) ).isEqualTo( "Hello-World" );
+        assertThat( Strings.join( Arrays.asList( "Hello" ), "-" ) ).isEqualTo( "Hello" );
+    }
+
+    @Test
+    public void testJoinAndCamelize() throws Exception {
+        assertThat( Strings.joinAndCamelize( new ArrayList<String>() ) ).isEqualTo( "" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "World" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "world" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "hello", "world" ) ) ).isEqualTo( "helloWorld" );
+    }
+
+    @Test
+    public void testIsEmpty() throws Exception {
+        assertThat( Strings.isEmpty( null ) ).isTrue();
+        assertThat( Strings.isEmpty( "" ) ).isTrue();
+        assertThat( Strings.isEmpty( " " ) ).isFalse();
+        assertThat( Strings.isEmpty( "not empty" ) ).isFalse();
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithArrayExistingVariables() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]" ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends" ) ).isEqualTo( "extends_" );
+        assertThat( Strings.getSaveVariableName( "class" ) ).isEqualTo( "class_" );
+        assertThat( Strings.getSaveVariableName( "Class" ) ).isEqualTo( "class_" );
+        assertThat( Strings.getSaveVariableName( "Case" ) ).isEqualTo( "case_" );
+        assertThat( Strings.getSaveVariableName( "Synchronized" ) ).isEqualTo( "synchronized_" );
+        assertThat( Strings.getSaveVariableName( "prop", "prop", "prop_" ) ).isEqualTo( "prop__" );
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithCollection() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]", new ArrayList<String>() ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends", new ArrayList<String>() ) ).isEqualTo( "extends_" );
```

Comment:
That would be nice! Doesn't have to be now, but I'd like it. Not necessarily with the underscore before the number, and maybe leave out the leading zeros (implementation should be the same effort).
But there are also some spots where we could derive a more meaningful default name as well, e.g. for result beans in methods that return `List<NodeDTO>` we simply use `list` which might then be conflicting already. Same for some of the setter wrappers, e.g.:
```java
// should be named 'children'
        List<NodeDTO> list = nodeListToNodeDTOList( node.getChildren() ); 
        if ( list != null ) {
            nodeDTO.setChildren( list );
        }
// should be named 'attributes'
        List<AttributeDTO> list_ = attributeListToAttributeDTOList( node.getAttributes() );
        if ( list_ != null ) {
            nodeDTO.setAttributes( list_ );
        }
```